core/sail/sail-spin/src/main/java/org/eclipse/rdf4j/sail/spin/SpinSail.java

```diff
@@ -29,6 +29,23 @@
 import java.util.ArrayList;
 import java.util.List;
 
+/**
+ * <p>
+ * The SpinSail is currently deprecated. If you are using SPIN to validate your data then it is recommended to move to
+ * SHACL with the ShaclSail. SHACL does not support any of the SPARQL inference features of SPIN, however some were
+ * proposed in "SHACL Advanced Features" but were not made a part of the W3C Recommendation in 2017.
```

Comment:
I'd rephrase this:

 "SHACL does not support any of the SPARQL inference features of SPIN, however some were proposed in "SHACL Advanced Features" but were not made a part of the W3C Recommendation in 2017." 

to something shorter, like: "Currently, the <a href="https://www.w3.org/TR/shacl/">SHACL W3C Recommendation</a> only supports validation, and has no equivalent to SPIN's inference features."